src/routers/router.rs

```diff
@@ -10,6 +10,22 @@ use matchit::Node;
 
 use anyhow::{bail, Error, Result};
 
+#[derive(Debug)]
+pub enum RouteType {
+    BeforeRequest,
+    AfterRequest,
+}
+
+impl RouteType {
+    pub fn from_str(input: &str) -> RouteType {
+        match input {
+            "BEFORE_REQUEST" => RouteType::BeforeRequest,
+            "AFTER_REQUEST" => RouteType::AfterRequest,
+            _ => panic!("Invalid route type enum."),
+        }
+    }
+}
+
```

Comment:
These routes are just for the middleware router and not the global router. So, I think something like `MiddlewareRouteType` would be more relevant.

Also, we usually create a `types` folder in the namespace. e.g. `routes/types.py`

Reply:
updated PR with given details. Thanks for pointing it out.